app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt

```diff
@@ -67,18 +71,29 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                     }
                 }
                 is PasscodeViewModel.PasscodeSetup -> {
-                    if (ownerImported) {
-                        findNavController().popBackStack(R.id.ownerAddOptionsFragment, true)
+                    if (BiometricManager.from(requireContext())
+                            .canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_STRONG or BiometricManager.Authenticators.BIOMETRIC_WEAK) == BiometricManager.BIOMETRIC_SUCCESS
+                    ) {
+                        val dialogBinding = DialogEnableBiometryBinding.inflate(LayoutInflater.from(context), null, false)
+                        dialogBinding.message.setText(R.string.settings_passcode_enable_biometry)
```

Comment:
this dialog will be called every time when changing some passcode setting, even if biometrics is already setup

Reply:
Not on every action but on every passcode change. And that is not necessary. You are right.